### PR TITLE
Adds admission reason to Booking and move `INFERRED_RELEASE` and `REMOVED_FROM_SOURCE` into CustodyStatus

### LINKED ISSUE #536: There should be an `admission_reason` field in booking
Let's start with an enum with values:

From Escape
New Commitment
Parole Violation
Probation Violation
Transfer

### PR DESCRIPTION
**Pull request by terinpw**
Closes #536

### CODE REVIEW

File: recidiviz/common/constants/booking.py [hunk 1]
```diff
@@ -21,6 +21,18 @@
 from recidiviz.common.constants.mappable_enum import MappableEnum
 
 
+class AdmissionReason(MappableEnum):
+    FROM_ESCAPE = enum_strings.admission_reason_from_escape
```
Comment (zdg2102) [inline recidiviz/common/constants/booking.py]: nit: can we just call this `ESCAPE`?

File: recidiviz/common/constants/enum_canonical_strings.py [hunk 2]
```diff
@@ -58,23 +58,29 @@
 
 # Booking
 
+admission_reason_from_escape = 'FROM_ESCAPE'
```
Comment (zdg2102) [inline recidiviz/common/constants/enum_canonical_strings.py]: Same here

File: recidiviz/ingest/models/ingest_info.proto [hunk 3]
```diff
@@ -48,6 +48,7 @@ message Person {
 message Booking {
 	optional string booking_id = 1;
 	optional string admission_date = 2;
+    optional string admission_reason = 14;
```
Comment (zdg2102) [inline recidiviz/ingest/models/ingest_info.proto]: nit: this doesn't seem properly indented

File: recidiviz/common/constants/booking.py (hunk 1 above)
Comment (terinpw) [reply recidiviz/common/constants/booking.py]: @ohinds to comment -- took it directly from issue #536, but that would work for me.

File: recidiviz/ingest/models/ingest_info.proto (hunk 3 above)
Comment (terinpw) [reply recidiviz/ingest/models/ingest_info.proto]: was a tab/space issue that looked fine on my machine. Updated file to use spaces as that's what our codebase does.

**Pull request merged**